.github/workflows/run_benchmarks_main.yml

```diff
@@ -0,0 +1,51 @@
+name: Run benchmarks main
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  run_benchmarks:
+    name: Run benchmarks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+
+      - name: Checkout main branch
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          path: main
+
+      - name: Install dependencies
+        run: make install
+        working-directory: ./main
+
+      - name: Run benchmarks
+        run: pytest --benchmark-only --benchmark-json benchmarks.json
+        working-directory: ./main
+
+      - name: Extract benchmarks
+        run: python scripts/extract_benchmarks.py -i benchmarks.json -o benchmarks.csv -t ubuntu-latest
+        working-directory: ./main
+
+      - name: Checkout benchmarks-data branch
+        uses: actions/checkout@v4
+        with:
+          ref: benchmarks-data
+          path: benchmarks-data
+
+      - name: Add and commit benchmarks data
+        run: |
+          uuid=$(uuidgen)
+          cp ../main/benchmarks.csv "uncollected/$uuid.csv"
```

Comment:
UUID is used as the filename here. The filename isn't important (we'll collect the CVSs using a glob), so long as it is unique.